src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -307,9 +307,12 @@ private boolean isTypeArgumentExplicit(TypeReference[][] typeArguments) {
 	 */
 	CtReference getDeclaringReferenceFromImports(char[] expectedName) {
 		CompilationUnitDeclaration cuDeclaration = this.jdtTreeBuilder.getContextBuilder().compilationunitdeclaration;
+		if (cuDeclaration == null) {
```

Comment:
what's the added value of this change?

Reply:
The previous check `cuDeclaration != null` makes no sense. It was performed too late:

```
LookupEnvironment environment = cuDeclaration.scope.environment; // <= cuDeclaration dereference (if cuDeclaration is null, we will get NullPointerException)
if (cuDeclaration != null && cuDeclaration.imports != null) { ... } // <= cuDeclaration could not be null here
```

Alternatively, I can remove that check at all, if `this.jdtTreeBuilder.getContextBuilder().compilationunitdeclaration` could not be null.